### DISCUSSION TITLE
Hello, why doesn't the right-click menu pop up

### DISCUSSION THREAD

**Discussion by ddkwork** [cogentcore/core#826] [Q&A] [answered]
```go
func vstart() {
	b := gi.NewAppBody("vstart")
	b.Style(func(s *styles.Style) {
		s.Direction = styles.Row
		s.Display = styles.Grid
		//s.Overflow.Set(styles.OverflowAuto)
	})
	for i := 0; i < 300; i++ {
		gi.NewButton(b).SetText(fmt.Sprint(i)).AddContextMenu(func(m *gi.Scene) {
			gi.NewButton(m).SetText("run")
			gi.NewButton(m).SetText("open dir")
		})
	}
	b.NewWindow().Run().Wait()
}

```

**Comment by rcoreilly** [comment]
Interesting -- the Button has its own Menu field, but probably we should just use the standard ContextMenu functionality.

**Comment by kkoreilly** [comment, marked as answer]
It should be fixed now (in 20144e530e8b997de3f3b1cb949550c81c2025c1). Thanks for reporting this!

**Comment by rcoreilly** [comment]
Kai fix doesn't actually use the ContextMenus, right?  What about changing button to use that instead of a separate Menu field?

**Reply by kkoreilly** [reply]
You don't understand what `Button.Menu` is. It is a menu displayed on click, not on right click, so it is not a context menu.

**Reply by rcoreilly** [reply]
yeah, but in the context of a button, would you ever have both a context menu and a click menu?  your fix precludes that by showing the regular Menu instead of the context menu?  If we don't want to allow both context and click menu, then it should just use context menu.  The advantage of the context menu is that it now has the stack, whereas button menu is singular, so it makes it more performant.  and simpler to not have 2 menus on one thing.

**Reply by kkoreilly** [reply]
I did exactly the opposite of that; I made it so that you can have both a click and a context menu.
